### PR TITLE
Update Dependencies #5

### PR DESCRIPTION
**Pull request by genericlady**
- ~Update Elixir from 1.2 to 1.4~

- Change Bamboo dependency github url in Mix.exs
  to thoughtbot/bamboo and master branch. We are
  doing this because we want to only have the most
  recent stable version of bamboo as a dependency.

- Add Sparkpost Elixir project to list of dependencies
  since we need this for transmitting an email to
  a Sparkpost endpoint.

- Remove Comments from Mix.exs since we don't need them.
  Clean up any code smells and side affects.

- Explicitly invoke base_uri inside of SparkpostAdapter as a
  function to eliminate side affect messages warning us about
  no variable found expanding to function.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -1,23 +1,24 @@
 defmodule Bamboo.SparkPostAdapter.Mixfile do
   use Mix.Project
 
-  @project_url "https://github.com/andrewtimberlake/bamboo_sparkpost"
+  @project_url "https://github.com/sparkpost/bamboo_sparkpost"
   @version "0.5.1"
```
Comment (richleland) [inline mix.exs]: I saw the original author decided to follow the version of bamboo. Thoughts on this? I think that ties the library to specific releases of bamboo too much, but not sure if this is an elixir convention or what. I'm thinking we should update to 0.6.0-alpha.1 for this round of work we'll be doing.

File: mix.exs [hunk 2]
```diff
@@ -42,18 +40,10 @@ defmodule Bamboo.SparkPostAdapter.Mixfile do
...
-  #
-  # Or git/path repositories:
-  #
-  #   {:mydep, git: "https://github.com/elixir-lang/mydep.git", tag: "0.1.0"}
-  #
-  # Type "mix help deps" for more examples and options
   defp deps do
     [
-      {:bamboo, github: "andrewtimberlake/bamboo", branch: "feature/attachments"},
+      {:bamboo, github: "thoughtbot/bamboo", branch: "master"},
```
Comment (richleland) [inline mix.exs]: Do we want to depend on master? That would mean this library is susceptible to upstream changes at any time. The current version is 0.8.0.

**Review by richleland — COMMENTED** [review]
Left a few comments/questions for ya.

File: mix.exs (hunk 1 above)
Comment (genericlady) [reply mix.exs]: Changed @version to 0.6.0-alpha.1

File: mix.exs (hunk 2 above)
Comment (genericlady) [reply mix.exs]: explicitly changed bamboo dependency version `{:bamboo, "~> 0.8.0"}`

Comment (genericlady) [reply mix.exs]: Now getting a side affect warning after running Mix deps.get but I don't think it's a big issue. Tests still pass.

```
warning: function Phoenix.View.render_to_string/3 is undefined (module Phoenix.View is not available)
Found at 2 locations:
  lib/bamboo/phoenix.ex:271
  lib/bamboo/phoenix.ex:281
```

**Review by richleland — APPROVED** [review]
lgtm - :shipit:

**Pull request merged**